richtext/tags.lua

```diff
@@ -65,9 +65,29 @@ end)
 
 M.register("img", function(params, settings)
 	local texture, anim = params:match("(.-):(.*)")
+	local width, height
```

Comment:
I think this looks a bit messy. I would suggest the addition of a helper function to split strings on first occurrence of a token. Like this:

```


-- Split string at first occurrence of token
-- If the token doesn't exist the whole string is returned
-- @param s The string to split
-- @param token The token to split string on
-- @return before The string before the token or the whole string if token doesn't exist
-- @return after The string after the token or nul
local function split(s, token)
	if not s then return nil, nil end
	local before, after = s:match("(.-)" .. token .. "(.*)")
	before = before or s
	return before, after
end
```

And for the img tag it would be used like this:

```
	local texture_and_anim, params = split(params, ",")
	local width, params = split(params, ",")
	local height, params = split(params, ",")
	
	local texture, anim = split(texture_and_anim, ":")
	width = width and tonumber(width)
	height = height and tonumber(height) or width
```